azure/functions/sql.py

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import collections.abc
+import json
+import typing
+
+from azure.functions import _sql as sql
+
+from . import meta
+
+
+class SqlConverter(meta.InConverter, meta.OutConverter,
+                   binding='sql'):
+
+    @classmethod
+    def check_input_type_annotation(cls, pytype: type) -> bool:
+        return issubclass(pytype, sql.SqlRowList)
+
+    @classmethod
+    def check_output_type_annotation(cls, pytype: type) -> bool:
+        return issubclass(pytype, (sql.SqlRowList, sql.SqlRow))
+
+    @classmethod
+    def decode(cls,
+               data: meta.Datum,
+               *,
+               trigger_metadata) -> typing.Optional[sql.SqlRowList]:
+        if data is None or data.type is None:
+            return None
+
+        data_type = data.type
+
+        if data_type in ['string', 'json']:
+            body = data.value
+
+        elif data_type == 'bytes':
+            body = data.value.decode('utf-8')
+
+        else:
+            raise NotImplementedError(
+                f'unsupported queue payload type: {data_type}')
+
+        rows = json.loads(body)
+        if not isinstance(rows, list):
+            rows = [rows]
+
+        return sql.SqlRowList(
+            (None if row is None else sql.SqlRow.from_dict(row))
+            for row in rows)
+
+    @classmethod
+    def encode(cls, obj: typing.Any, *,
+               expected_type: typing.Optional[type]) -> meta.Datum:
```

Comment:
where is expected_type used?

Reply:
Not sure what exactly this is used for but I see that all the other binding types encode function also have this. Ex. https://github.com/Azure/azure-functions-python-library/blob/3bf08e13539ba957c7df4230fc9b27fc860486fe/azure/functions/cosmosdb.py#L54